src/realm/obj.cpp

```diff
@@ -44,6 +44,133 @@
 #include "realm/util/overload.hpp"
 
 namespace realm {
+namespace {
+
+struct Helper {
```

Comment:
Can we call it something more descriptive? Maybe "NestedCollectionIndex"?

Reply:
Changed to `NestedCollectionInfo`.